Development

TarasTsalko opened the pull request.

Commit:
Реализован базовый pipline для отрисовки фрактала

Commit:
Добавлен процесс отрисовки и реализован процесс перерисовки при изменении
размера окна

Commit:
Реализован Zoom

Commit:
Добавлен класс WaitForFPS

Commit:
Добавлены модульные тесты для MandelbrotSender

Commit:
Убрана неиспользуемая функция

Commit:
Добавлены модульные тесты и обработка клавиши Esc

Commit:
Добавлены модульные тесты и убран избыочный код

Commit:
Добавлены коментарии

mamontov-cpp commented on include/sfml_events_handler.hpp:
```diff
@@ -56,17 +104,47 @@ class SfmlEventHandler {
+            state_.viewport.y_max = target_y + new_height / 2.0;
+
+            // Дополнительные проверки границ, чтобы не выйти за пределы множества Мандельброта
+            if (state_.viewport.x_min < -2.5)
```
Здесь бы и границы лучше вынести в константы. Ну и можно просто применить std::min/std::max для обрезания значений, обычно в графике так делается.

TarasTsalko replied:
Исправил. Вынес в константы границы, переписал код с использованием std::min/std::max.

mamontov-cpp commented on src/main.cpp:
```diff
@@ -27,12 +34,46 @@ class FrameClock {
+        // Рассчитываем максимально допустимую длительность кадра
+        // 1000 мс делим на желаемую частоту кадров (fps_)
+        // Например, при fps_ = 60 получаем 16.67 мс на кадр
+        const auto max_frame_duration = std::chrono::milliseconds(1000 / fps_);
```
1000 бы тоже вынести в константу.

TarasTsalko replied:
Исправил. Вынес в константу.

mamontov-cpp commented:
Принято с замечаниями. Хорошая работа, но замечания лучше всё же исправить.

Commit:
Исправлены замечания по ревью, добавлены необходимые константы.
Код с if-ми для проверки границ заменён на код с использованием std::min/std::max

mamontov-cpp commented:
Увидел, спасибо!
